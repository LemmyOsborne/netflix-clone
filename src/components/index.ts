export { Jumbotron } from "./Jumbotron/Jumbotron";
export { Footer } from "./Footer/Footer";