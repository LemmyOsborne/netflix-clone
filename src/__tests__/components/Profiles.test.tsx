import React from "react";
import { render } from "@testing-library/react";
import { Profiles } from "../../components";

describe("<Profiles/>", () => {
    it("renders <Profiles/>", () => {
        const { container, getByText, getByTestId } = render(
            <Profiles>
                <Profiles.Title>Who's watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.User onClick={() => {}}>
                        <Profiles.ProfileImage src={`/images/users/2.png`} alt="Profile Image" data-testid="profile-image" />
                        <Profiles.Name>Lemmy</Profiles.Name>
                    </Profiles.User>
                </Profiles.List>
                <Profiles.Button>Manage Profiles</Profiles.Button>
            </Profiles>
        )

        expect(getByText("Who's watching?")).toBeTruthy()
        expect(getByText("Lemmy")).toBeTruthy()
        expect(getByText("Manage Profiles")).toBeTruthy()
        expect(getByTestId("profile-image")).toBeTruthy()
        expect(container.firstChild).toMatchSnapshot()
    })
})