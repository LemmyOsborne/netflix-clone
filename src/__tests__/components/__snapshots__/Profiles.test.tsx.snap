// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles/> renders <Profiles/> 1`] = `
<div
  class="sc-iNGGcK gptbzR"
>
  <h1
    class="sc-jeraig ciXKSy"
  >
    Who's watching?
  </h1>
  <ul
    class="sc-eJwWfJ lnpsVZ"
  >
    <li
      class="sc-ehCJOs iRMOTV"
    >
      <img
        alt="Profile Image"
        class="sc-nVkyK irqvPW"
        data-testid="profile-image"
        src="/images/users/2.png"
      />
      <p
        class="sc-hiwPVj fOkDCV"
      >
        Lemmy
      </p>
    </li>
  </ul>
  <button
    class="sc-lcepkR hQGDvq"
  >
    Manage Profiles
  </button>
</div>
`;
